frontend/src/components/pages/events/CountdownButton.tsx

```diff
@@ -107,6 +95,78 @@ export const CountdownButton: React.FC<Props> = ({
     } ${translate(timeparts[1], timeLeft[timeparts[1]])}`;
   };
 
+  if (currentTimeParts.length !== 0) return <>{getCurrentTimeLeft(currentTimeParts)}</>;
+  if (isSignedUp) return <>Meld av</>;
+  if (isOnWaitingList) {
+    if (positionOnWaitinglist === 2) {
+      return (
+        <>
+          Det er en person foran deg i ventelisten
+          <br /> Trykk her for å melde av
+        </>
+      );
+    } else if (positionOnWaitinglist === 1) {
+      return (
+        <>
+          Du er på første plass i ventelisten <br />
+          Trykk her for å melde av
+        </>
+      );
+    } else
+      return (
+        <>
+          Det er {positionOnWaitinglist - 1} personer foran deg i ventelisten
+          <br /> Trykk her for å melde av
+        </>
+      );
+  }
+
+  if (isFull) return <>Meld på venteliste</>;
+  return <>Meld På</>;
+};
+
+/**
+ * Component for the count down button on the detail page of an attendable event
+ *
+ * Props:
+ * - countDownDate: the date that is counted down to
+ * - currentTime: the time right now
+ * - isSignedUp: whether the user viewing the page is signed up to the event
+ * - isOnWaitingList: whether the user viewing the page is on the waiting list for the event
+ * - isFull: whether the event is full (all available slots are taken)
+ * - loading: whether the button should show a loading symbol
+ * - disabled: whether the button should be disabled
+ * - onClick: metod called when the count down button is clicked
+ * - styleClassName: styled class
+ */
+
+export const CountdownButton: React.FC<Props> = ({
+  countDownDate,
+  currentTime,
+  isSignedUp,
+  isOnWaitingList,
+  positionOnWaitinglist,
+  isFull,
+  loading,
+  disabled,
+  onClick,
+}) => {
+  const [now, setNow] = useState(dayjs(currentTime));
+  const [timeLeft, setTimeLeft] = useState(calculateTimeLeft(countDownDate, now));
+
+  useEffect(() => {
+    // Update timeLeft and now (current time) each second
+    const id = setTimeout(() => {
+      setTimeLeft(calculateTimeLeft(countDownDate, now));
+      setNow(now.add(1, "second"));
+    }, 1000);
+    return () => {
+      clearTimeout(id);
+    };
+  });
```

Comment:
It seems to me that this logic is duplicated, both in `ButtonText` and here in `CountdownButton`. Try evaluate if it's needed in both places or just one of them, and if you need it both places, pass it as a prop rather than duplicating the logic 😊